cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/ConnectorSourceEmitter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.batch;
+
+import co.cask.cdap.etl.api.ErrorRecord;
+import co.cask.cdap.etl.batch.mapreduce.ErrorOutputWriter;
+import co.cask.cdap.etl.common.RecordInfo;
+import com.google.common.collect.Multimap;
+
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Just like a PipeEmitter, except it doesn't need to add stage name itself, since the ConnectorSource does it.
+ */
+public class ConnectorSourceEmitter extends PipeEmitter {
+
+  private ConnectorSourceEmitter(String stageName,
+                                 Set<PipeStage<RecordInfo>> outputConsumers,
+                                 Multimap<String, PipeStage<RecordInfo>> outputPortConsumers,
+                                 Set<PipeStage<RecordInfo<ErrorRecord<Object>>>> errorConsumers,
+                                 @Nullable ErrorOutputWriter<Object, Object> errorOutputWriter) {
+    super(stageName, outputConsumers, outputPortConsumers, errorConsumers, errorOutputWriter);
```

Comment:
for my understanding, since `getPipeRecord` is not adding stageName, which one is this `stageName` passed in the ctor? Also it would be good to add an example in javadoc to make it clear.